### DISCUSSION TITLE
WAR Outside Temp not clear

### DISCUSSION THREAD

**Discussion by sabom2d** [edterbak/HeishaMoNR#136] [Q&A] [answered]
Hi ,
I have two HP running with Heishamon and Node-Red addon. An MD7 and an MD9 Version.
The MD7  is showing WAR Outside Temp correctly as 11Degrees Celsius as it is placed on the north side of the house
But the MD9 is showing 29 degrees ( it is placed on the south-east of the house.)
I would like to use the outside Temp of the MD7 also for the MD9 as I have done already for the Dash board setting.
I do not understand where I have to tweak the WAR Outside temp for the MD9.
What I found is 2 Outside entities in the WAR node:
![Node-red-Dash Control](https://github.com/edterbak/NodeRed_Heishamon_control/assets/107671129/f0484063-6716-46e6-b8d8-9a51c40fe049)

So, where do I have to change the node, so that the MD7 entity is used?

Thanks in advance

ede

**Comment by edterbak** [comment]
Hi, 
Wow. you have 2 HP's working AND both with the flow?
Do you have 2 instances of HomeAssistant with NodeRed Add-on? or 2 instances Node Red in a container e.g.? 
Curious curious 😄 

What is your configuration. Are both HP's setup with Direct operating mode or Compensation Curve (CC)?


In case you use both pumps in Direct mode, 
In the WP Control tab, on the top, little to the right, there is a section for that.
Maybe it looks slightly different on your setup, because I am working on a beta which changes this to make it easier
![image](https://github.com/edterbak/NodeRed_Heishamon_control/assets/3155621/45ab9160-3810-4883-8fef-e88c9b75112c)

If you connect your outside temperature input to the grey top link-in node, it should detect it and use the value. 
Make sure it is ONLY a payload with the temperature, and not mixed topic/value pairs. (this is the bit im trying to improve)

But, keep in mind, the WAR function inside node red will only work if you are using direct mode. 
The compensation curve is the same function, but then the one that panasonic created, with fixed sensor (internal/external). No option to connect any other source here. 

Does this help?? 😄

**Comment by sabom2d** [comment]
Hi Ed,

thanks for your fast response.
Yes I am running 2 HP, a 7kW anf a 9kW version, each one with Heishamon and your Node-Red addon in seperate Docker containers on a raspi. I also have Homeassistant running on another Raspi but not yet configured correctly for the HP. Up to yesterday both addons were running smooth, but just today both are struggeling, the MD7 Addon has Error messages like 
[error] [function:cee35b0a1e2da0ea] TypeError: Cannot read properties of undefined (reading 'function_active_state')
The MD9 though does not send any error , but is not working as expected either.
I have the feeling, I need to install both again.

But what I already learned by your questions is, that I do not really have an understanding how to use your addon.
So, 
1st Question for Background understanding: What is Direct mode? Do I need this to run the HP correctly?
2nd Question: what can I do against this error messages:
![Error_messagesMD7](https://github.com/edterbak/NodeRed_Heishamon_control/assets/107671129/b9d75d85-3659-4215-a940-37c5dce14ec6)

Thanks in advance

**Comment by edterbak** [comment]
**Direct mode / Compensation curve**
The options in the heatpump are (1) Direct Mode and (2) Compensation Curve (CC).
The Direct mode is a direct / manual setup. You directly enter the setpoint of the water on the controller, or if you use my flow, in the dashboard. My flow works well with this mode. 

The panasonic Compensation curve is nice. It is taking into account the outside temperature. If it is cold, it will increase the water temperature. If it is warm outside, it will automatically lower the water temperature. 

I have copied this functionality in the flow, because the panasonic CC function is fixed to the panasonic temperature sensor. I called it WAR-function. (stupid name, I know)
But my flow offers more flexibility on what sensor you want to use for outside temperature. 



My advice is to use the Direct mode setting. THis is what works best currently. 
At least until I fix the other option :)
![image](https://github.com/edterbak/NodeRed_Heishamon_control/assets/3155621/69547d99-eb21-4e43-9b68-d3bd81f41b55)


**Error messages**
The error message seems to suggest that the context store "file" is not set propperly. 
Did you follow the instruction to change settings.js and enable the persistent storage?


Side note:
You are running node red in docker on a pi.
Do you use a SSD or a small SD-card as storage?
I know the SD-cards typically wear out fast. 
The flow I have built, is not sparing on the read/write actions in the storage. It might become an issue at some point that the sd-card is corrupt.

regards,

**Comment by sabom2d** [comment]
Hi Ed,

thanks for explaining it again, I read the instructions but like I said, did not quite understand it correctly.
Yes I am running it in a docker on a PI, atm still in testing mode, since the weekend I am using the HP finally to also heating, before it was only for DHW. But your hint is good, I will change the setup later to storing on USB drives when I have configured everything.

Regarding persistent File, I now think, that I misread your instructions, I did not install docker node-red with the -V flag. I must admit, it's the first time I use docker and I'm not familiar yet with this solution.

Regarding HP Setup, I did not setup Direct mode yet, not sure how to do this. Will check the manual to get this setting. I also have setup variable Pump speed, do I ned to set this fixed also?

regards

**Comment by sabom2d** [comment]
Update:

1. I have set the HP to Direct Temperature mode
2. I have restarted node-red with persistent folder ( I hope ): 
    console command: node-red -u ~/node-red-MD9 -v FOLDER:/node-red-MD9-data

But still on the MD7 HP I'm getting the errors:
![Error_messagesMD7-2](https://github.com/edterbak/NodeRed_Heishamon_control/assets/107671129/c704ff60-9c7b-4759-b114-c999a6d4cf8c)

My WAR Outside Temp seems to be correct now on the MD9 as it is the same, as it is on the MD7 HP.
Still there is some difference in the 2 WAR layouts I don't understand:
![Compare-War-MD7-MD9](https://github.com/edterbak/NodeRed_Heishamon_control/assets/107671129/d780e36e-49fa-4d0f-a9ab-cf5703e8e903)

So at least , I have some progress, although I'm still not final.

regards

**Reply by sabom2d** [reply]
Update 2: 

looks like the difference was, because on MD7 DHW mode was active, not heating mode.
Now that I switched to Heating Mode, at least the Calculated Temp is visible.

I also found the root cause for the error messages, now as I read your instructions again and checked my config files again. In the MD7 settings file the contextStorage was not changed, maybe I forgot to save the changes. 

Again most issues were related to not reading and following the instructions as it seems.
I'm sorry that you had to spend time on an Error 80 ( 80cm in front of the keyboard ).
Thanks again for this fine peace of software. 
Next step is not to get it integrated into HA.

regards

ede

**Comment by edterbak** [comment, marked as answer]
HI, thats good news. So everything is now running fine. 
If so you can close the issue if you like. 

About the integration with Home Assistant I am not familiar of how to create a kiosk page with iframe and its limitations. Cant help there. 

If you are thinking about installing the custom HACS component for heishamon, you can do that, but you have to make sure that all automations created (automatically) for that are turned off. 
Otherwise corrections NR flow does, are countered by HA logic. 
A lot of MQTT traffic and a heatpump that is stressed could be the result. Just to keep in mind :)
